app/code/community/Ebizmarts/MailChimp/Model/Subscriber.php

```diff
@@ -30,10 +30,70 @@ public function sendConfirmationRequestEmail()
 
     public function sendConfirmationSuccessEmail()
```

Comment:
At this point the subscription to MailChimp should be sent and not before in order to avoid incoherence with Magento status.